### PR TITLE
Support zip artifacts for KC Build

### PR DESCRIPTION
**Pull request by sknot-rh**
Signed-off-by: Stanislav Knot <sknot@redhat.com>

### Type of change
- Enhancement / new feature

### Description
Fixes https://github.com/strimzi/strimzi-kafka-operator/issues/4178

https://github.com/strimzi/strimzi-kafka-operator/pull/4168 introduced building from tar.gz and jar artifacts. This PR adds support for zip artifacts. 
The difference between tar.gz and zip is that zip is able to pack symlinks what can lead to the security risk. That is the reason why all the symlinks are removed from the unpacked archive.

This PR can be tested with this resource:
https://pastebin.com/8pR4e52w

### Checklist
- [x] Write tests
- [x] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: api/src/main/java/io/strimzi/api/kafka/model/connect/build/Artifact.java [hunk 1]
```diff
@@ -36,11 +37,12 @@
 
     public static final String TYPE_JAR = "jar";
     public static final String TYPE_TGZ = "tgz";
+    public static final String TYPE_ZIP = "zip";
 
     private Map<String, Object> additionalProperties = new HashMap<>(0);
 
     @Description("Artifact type. " +
-            "Currently, the supported artifact types are `tgz` and `jar`.")
+            "Currently, the supported artifact types are `tgz`, `jar` and `zip`.")
```
Comment (scholzj) [inline api/src/main/java/io/strimzi/api/kafka/model/connect/build/Artifact.java:45]: I wonder if @PaulRMellor would ask for a coma before the `and`.

**Review by scholzj — COMMENTED** [review]
One nit. LGTM otherwise. Should we maybe use the ZIP also in one of the STs? Camel Connectors have ZIPs as well as TGZs. So maybe we can just add one more connector or change one of the existing to have all JAR, ZIP and TGZ covered in the STs without adding a new test dedicated to it.

File: api/src/main/java/io/strimzi/api/kafka/model/connect/build/Artifact.java (hunk 1 above)
Comment (PaulRMellor) [reply api/src/main/java/io/strimzi/api/kafka/model/connect/build/Artifact.java:45]: IBM style wants a comma before conjunction in a series of three or more :+1:

Comment (sknot-rh) [reply api/src/main/java/io/strimzi/api/kafka/model/connect/build/Artifact.java:45]: Thanks for the review. @im-konge is working on the STs in the https://github.com/strimzi/strimzi-kafka-operator/pull/4283 
I will talk to him, whether it makes sense to add `zip` tests into his PR (eventually after merging this).

File: api/src/main/java/io/strimzi/api/kafka/model/connect/build/ZipArtifact.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.api.kafka.model.Constants;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+/**
+ * TGZ artifact represents an artifact which is downloaded and unpacked using `unzip`
```
Comment (ppatierno) [inline api/src/main/java/io/strimzi/api/kafka/model/connect/build/ZipArtifact.java:15]: isn't it supposed to be ZIP?

**Review by ppatierno — APPROVED** [review]
Just one nit, lgtm otherwise.

File: api/src/main/java/io/strimzi/api/kafka/model/connect/build/ZipArtifact.java (hunk 2 above)
Comment (sknot-rh) [reply api/src/main/java/io/strimzi/api/kafka/model/connect/build/ZipArtifact.java:15]: Good catch, thanks

**Comment by scholzj** [comment]
> Thanks for the review. @im-konge is working on the STs in the #4283
I will talk to him, whether it makes sense to add zip tests into his PR (eventually after merging this).

Ahh, right. Forgot that is not merged yet.

**Review by scholzj — APPROVED** [review]

**Pull request merged**